--- sequence.py	2022-09-01 13:45:00.182081408 -0400
+++ sequence2.py	2022-09-01 14:15:10.213101140 -0400
@@ -42,6 +42,22 @@
 		self.seq.happened = True
 		return f'Character Driver Module {num}: write (number {self.file.num})'
 
+class llseek_op:
+	def __init__(self, file):
+		self.file = file
+		self.seq = sequencer()
+	def sequence(self, num):
+		self.seq.happened = True
+		return f'Character Driver Module {num}: llseek (number {self.file.num})'
+
+class ioctl_op:
+	def __init__(self, file):
+		self.file = file
+		self.seq = sequencer()
+	def sequence(self, num):
+		self.seq.happened = True
+		return f'Character Driver Module {num}: ioctl (number {self.file.num})'
+
 class init_op:
 	def __init__(self):
 		self.seq = sequencer()
@@ -88,6 +104,20 @@
 		op.seq.after(self.m_open.seq)
 		self.m_close.seq.after(op.seq)
 		return op
+	def ioctl(self):
+		assert self.m_open is not None
+		assert self.m_close is not None
+		op = ioctl_op(self)
+		op.seq.after(self.m_open.seq)
+		self.m_close.seq.after(op.seq)
+		return op
+	def llseek(self):
+		assert self.m_open is not None
+		assert self.m_close is not None
+		op = llseek_op(self)
+		op.seq.after(self.m_open.seq)
+		self.m_close.seq.after(op.seq)
+		return op
 
 
 def main():
@@ -109,17 +139,21 @@
 		file1.read(),
 		file1.read(),
 		file1.write(),
+		file1.ioctl(),
+		file1.ioctl(),
 		file1.write(),
 		file1_release := file1.release(),
 		file2_open := file2.open(),
 		file2.read(),
 		file2.read(),
+		file2.llseek(),
 		file2.write(),
 		file2_release := file2.release(),
 		file3_open := file3.open(),
 		file3.read(),
 		file3.write(),
 		file3.write(),
+		file3.llseek(),
 		file3.write(),
 		file3_release := file3.release(),
 	]
